torch/csrc/autograd/function.cpp

```diff
@@ -1,17 +1,27 @@
 #include "Python.h"
 #include "function.h"
 
-#include <string>
-
 #include "variable.h"
 #include "torch/csrc/jit/ir.h"
 #include "torch/csrc/autograd/grad_mode.h"
 #include "torch/csrc/autograd/functions/special.h"
 
+#include <string>
+#include <algorithm>
+#include <cstdint>
+#include <vector>
+
 namespace torch { namespace autograd {
 
 thread_local uint64_t Function::function_counter = 0;
 
+bool any_variable_requires_grad(const variable_list& variables) {
```

Comment:
Can you please put it in the header? It compiles to an extremely simple for loop, so it's likely to get inlined.